llama_stack/providers/utils/inference/openai_mixin.py

```diff
@@ -230,6 +236,24 @@ async def openai_chat_completion(
         """
         Direct OpenAI chat completion API call.
         """
+        if self.download_images:
+
+            async def _localize_image_url(m: OpenAIMessageParam) -> OpenAIMessageParam:
+                if isinstance(m.content, list):
+                    for c in m.content:
+                        if c.type == "image_url" and c.image_url and c.image_url.url and "http" in c.image_url.url:
+                            localize_result = await localize_image_content(c.image_url.url)
+                            if localize_result is None:
+                                raise ValueError(
+                                    f"Failed to localize image content from {c.image_url.url[:42]}{'...' if len(c.image_url.url) > 42 else ''}"
```

Comment:
Did you actually calculate that "Failed to localize image content from" is 38 so you need to truncate at > 42? 😆

Reply:
:laughing: i had a bug where data: urls were getting through and the exception had MB of base64 data in the message.

Reply:
the length of the surrounding text and truncation size are unrelated. the surrounding text is not truncated.